### PR TITLE
upgrade developers.md

### LINKED ISSUE #747: 0.4.0 on Freebsd 10.2 is not downloading blocks.
The program compiled, but can't fetch blocks. About 120+ Mb RAM is used. netstat shows connections coming through.

 ./siac status
Block:  6581df2b1dae4ab8755cc2092ee438555ee1c705ecc34743d853c5d625c522c4
Height: 0
Target: [0 0 20 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]

gdb report:

$ gdb ./siad 69498
GNU gdb 6.1.1 [FreeBSD]
Copyright 2004 Free Software Foundation, Inc.
GDB is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.
Type "show copying" to see the conditions.
There is absolutely no warranty for GDB.  Type "show warranty" for details.
This GDB was configured as "amd64-marcel-freebsd"...
Attaching to program: /usr/home/crypto/Sia/siad, process 69498
Reading symbols from /lib/libthr.so.3...done.
[New Thread 801ed1800 (LWP 100610/siad)]
[New Thread 801e85c00 (LWP 100316/siad)]
[New Thread 801e85800 (LWP 100315/siad)]
[New Thread 801c6f800 (LWP 100314/siad)]
[New Thread 801c06c00 (LWP 100313/siad)]
[New Thread 801c06800 (LWP 100312/siad)]
[New Thread 801c06400 (LWP 100335/siad)]
Loaded symbols for /lib/libthr.so.3
Reading symbols from /lib/libc.so.7...done.
Loaded symbols for /lib/libc.so.7
Reading symbols from /libexec/ld-elf.so.1...done.
Loaded symbols for /libexec/ld-elf.so.1
[Switching to Thread 801ed1800 (LWP 100610/siad)]
runtime.sys_umtx_op () at /usr/local/go/src/runtime/sys_freebsd_amd64.s:45
45		MOVL	AX, ret+32(FP)
(gdb) bt
#0  runtime.sys_umtx_op () at /usr/local/go/src/runtime/sys_freebsd_amd64.s:45
#1  0x00000000004341a2 in futexsleep () at /usr/local/go/src/runtime/os_freebsd.c:78
#2  0x000000c2080215d8 in ?? ()
#3  0x000000000000000f in ?? ()
#4  0x0000000000000000 in ?? ()
Current language:  auto; currently minimal

### PR DESCRIPTION
**Pull request by DavidVorick**
fixes #747 

Some of the conventions are a little out of date. I've updated what I thought was important.

### CODE REVIEW

File: doc/Developers.md [hunk 1]
```diff
@@ -3,22 +3,34 @@ Developer Environment
...
+network. To get the release build, it is usually sufficient to run
+`go get -u github.com/NebulousLabs/Sia/...`. This will pull all of the
+necessary repositories off of github and place them in the correct folders
+within the gopath.
+
+If you would like to build the project manually, you will need to clone the Sia
+repository into the folder `$GOPATH/src/github.com/NebulousLabs/`. It will not
+be sufficient to close Sia from the gopath. It also seems that more recent
+versions of go do not like sym links. After cloning, you will need to run 
+`make dependencies`.
```
Comment (lukechampine) [inline doc/Developers.md]: "close" -> "clone"? What do you mean about symlinks?

You can, in fact, develop on Sia after running `go get`. You just need to modify your `.git/config`.

File: doc/Developers.md [hunk 2]
```diff
@@ -122,24 +151,21 @@ forkBlockchain(node)
...
+prefix to the name (explained below). The responsibility for thread-safety
+comes from the exported funcitons which call the non-exported functions.
+Maintaining this convention minimizes developer overhead when working with
+complex objects.
 
 Functions prefixed 'threaded' (example 'threadedMine') are meant to be called
-in their own goroutine ('go threadedMine()') and will manage their own mutexes.
-These functions typically loop forever, either listening on a channel or
-performing some regular task, and should not be called with a mutex locked.
+in their own goroutine ('go threadedMine()') and will manage their own
```
Comment (lukechampine) [inline doc/Developers.md]: backticks

File: doc/Developers.md [hunk 3]
```diff
@@ -189,41 +215,24 @@ to go unnoticed.
...
-unmarshalled without error.
-
-Compatibility
--------------
+The test suite code should be the same quality as the rest of the codebase.
+When writing new code in a pull request, the pull request should include test
+coverage for the code.
+
+Most modules have a tester object, which can be created by calling
+`createModuleTester`. Module testers typically have a consensus set, a miner, a
```
Comment (lukechampine) [inline doc/Developers.md]: potentially confusing, since `createModuleTester` is not a real function

File: doc/Developers.md [hunk 4]
```diff
@@ -189,41 +215,24 @@ to go unnoticed.
...
+wallet, and a few other relevant modules that can be used to build
+transactions, mine blocks, etc.
 
-Upgrades must preserve compatibility with previous versions. While Sia is still
-in beta, compatibility must be preserved until 90% no longer need the
-compatibilty adjustments.
+In general, testing that uses exclusively exported functions the achieve full
```
Comment (lukechampine) [inline doc/Developers.md]: the -> to

File: doc/Developers.md (hunk 1 above)
Comment (DavidVorick) [inline doc/Developers.md]: I'm 95% sure that if you use symlinks to put the github repo in your gopath, go will be unable to find the packages it needs

Comment (DavidVorick) [inline doc/Developers.md]: If you want to add a paragraph explaining the `.git/config` stuff, you can do that in another PR.

**Comment by DavidVorick** [comment]
typos fixed

File: doc/Developers.md (hunk 1 above)
Comment (lukechampine) [inline doc/Developers.md]: do you mind if I just push to this branch?

Comment (DavidVorick) [inline doc/Developers.md]: Yeah. I only said another PR because I was still working on this one, but I'm done now

**Comment by lukechampine** [comment]
ok, I *really* need to get to the bottom of that one

**Comment by DavidVorick** [comment]
might be sufficient to start vendoring in go 1.5

**Comment by lukechampine** [comment]
lol that's the muxado error. This one was the renter error
